### PR TITLE
Version docs

### PR DESCRIPTION
**Pull request by sprice**
This PR is by no means complete. Right now it's a simple proof of concept looking for some initial feedback.

We should have versioned docs so that when users are relying on documentation they can be certain they see the full and complete docs for the version of video.js they are working with.

Right now the docs are created against the `master` branch of video.js. There are a couple of reasons this isn't a good thing. One, unless `master` is integrated into `gh-pages` only when a new version is cut, the docs are not a correct representation of the latest version of video.js. Second and more importantly, there's not historical record of docs. If I'm still working on `4.x` there's no guarantee that what I'm looking at is correct. 

I've hacked together a method to capture the version number to use. Right now it's set in two places which sort of sucks (the Gruntfile for the git checkout and the file writing, and the client side JS via `vjs-version.js` for link building). I hope there's a better way to do this.

The current changes in this PR are that it does a `git checkout TAG` to get a specific version and then builds the docs within a `docs/VERSION/` directory rather than `docs/`. There's a bunch left to do like fixing up some links, advertising the version on the docs pages, and linking to the correct Github source code. If we agree this is worth doing I'm happy to finish this off.

### CODE REVIEW

**Comment by rcrooks** [comment]
I really like the idea of capturing version history - will take a look to see how this is progressing

**Comment by rcrooks** [comment]
@sprice this is looking brilliant! We'll need to build in some kind of selector for versions, but I think it may be adequate to do that just on the API index page -- that content is generated out of a separate client-side script (which I'd planned to port into the grunt process at some point anyway).

**Comment by gkatsev** [comment]
We also probably wouldn't want to check the docs/api into master but only on the gh-pages (our deploy branch).

**Comment by rcrooks** [comment]
So if we always run the grunt process in the gh-pages branch, that will 
take care of that, right? Sorry for the question, but I've really only 
done very basic things with git.

Also wanted to note for Shawn: your notes mentioned versioning back into 
4.x versions. It would be great if we could do that, but note that the 
current doc generator was written for 5.x, and hasn't been tested on 4.x 
versions at all -- I have no reason to think it would break, but just 
letting you know.

Gary Katsevman wrote:

**Comment by gkatsev** [comment]
Yeah, I believe we should be able to just run grunt when we merge master into gh-pages.

The 4.x docs are available in the videojs repo, we should probably just grab the latest of that to keep it around here with a 4.x/legacy link and not actually bother trying to built it like all the 5.x docs.

File: Gruntfile.js [hunk 1]
```diff
@@ -1,20 +1,28 @@
 'use strict';
 module.exports = function (grunt) {
+
+    // Set the video.js version
+    var vjsVersion = '5.2.1';
```
Comment (gkatsev) [inline Gruntfile.js]: this should probably be passed in as an argument to grunt or grab it from the package.json of videojs.

**Comment by sprice** [comment]
This is ready for an initial review.

On the docs site we have two "index" pages. [1](http://docs.videojs.com/) and [2](http://docs.videojs.com/docs).

The changes in this Pull Request bypass `/docs`. Right now this page doesn't work if it's manually navigated to. The `index.html` file is still there but I'm not sure what should be here in its place. Maybe a simple `index.html` page and javascript that navigates back to `/`?

File: Gruntfile.js [hunk 2]
```diff
@@ -1,20 +1,31 @@
...
             },
             cloneVideoJS: {
-                // Once 5.0 is in stable the line below should be use instead
-                // command: 'rm -rf ./video.js && git clone -b stable --single-branch https://github.com/videojs/video.js.git'
-                command: 'rm -rf ./video.js && git clone -b stable --single-branch https://github.com/videojs/video.js.git'
+                command: 'rm -rf ./video.js && git clone https://github.com/videojs/video.js.git'
             },
+            gitPull: {
+                // Used if video.js has alredy been cloned and we want the latest code to work with.
+                command: 'cd video.js && git pull origin master'
```
Comment (sprice) [inline Gruntfile.js:21]: Come to think of it I think this should be `git fetch`. Should it be `git fetch --tags`?

**Comment by rcrooks** [comment]
Hey Shawn,
Sorry, I missed this one. There are some conflicts now, so I looking to see...
Robert

**Comment by rcrooks** [comment]
Shawn,
I'm testing locally on you PR, but something is broken. Getting a blank (api) index page, and the individual API pages only show the head section and the navigation.

**Comment by sprice** [comment]
I just merged `master` into this PR. Things seem to be working again.

This PR now needs to add the ability to version the guides and examples.

I notice that we're generating the guides and examples and storing those files in `master`, while we ignore the generated api docs from `master` and generate them in `gh-pages`. Do we want to review that to keep things consistent? I kind of like having all the generated files in `master`.

**Comment by rcrooks** [comment]
I'll have another look.

I agree with you that it's nice having all the generated docs in master, but will defer to @gkatsev on that.

**Comment by rcrooks** [comment]
@sprice sorry, just finally checking this out after the holiday break. Docs look good. There's an issue with the API index page (can't find the vjsVersions var) but that can be worked out one way or another: either I can port the script that generates the index into the grunt file or just extract the version from the path. The former is the right thing to do, but I've put it off because it's a fair amount of work.

**Comment by sprice** [comment]
@rcrooks just wanted to let you know I'm a new dad and don't have as much free time as I once did. Please keep this PR open and I'll try to work on it when I get a chance but it may be some time.

**Comment by rcrooks** [comment]
@sprice Congratulations! and no problem....btw,  I did a kind of quick and dirty versioning for the Brightcove Player API docs -- just generated multiple parallel directory trees with the different versions of the docs, and then injected a version selector on the API index pages. It doesn't give you easy switching between versions at the class level, but I'm not sure how many people really need that.

**Comment by gkatsev** [comment]
For 6.x we're generating docs in video.js, so, this is obsolete in the current form. Thanks for the work so far, though, @sprice.
Hopefully, it'll be easier to do in the future (especially since the template we're using is working on adding versioning support).

**Pull request closed without merging**